frontend/app/utils/date-utils.ts

```diff
@@ -51,3 +54,73 @@ export function getLocalizedMonths(
     text: formatter.format(Date.UTC(0, i, 1)), // formatted month name
   }));
 }
+
+/**
+ * Validates whether a given string is in the format YYYY-MM-DD.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD, otherwise `false`.
+ *
+ * @example
+ * ```javascript
+ * const validDate = "2023-05-20";
+ * const invalidDate = "20-05-2023";
+ *
+ * console.log(isValidDateString(validDate)); // true
+ * console.log(isValidDateString(invalidDate)); // false
+ * ```
+ */
+export const isValidDateString = (dateString: string) => {
+  return isValid(parseISO(dateString));
+};
+
+/**
+ * Validates whether a given string is in past or today.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD and the date is in past or today's date, otherwise `false`.
+ *
+ * This function first validates the input date string using the `isValidDateString` function.
+ * If the date string is invalid, an error is thrown. Otherwise, it compare date
+ * object with the date string set to the start of the day in UTC (00:00:00.000Z).
+ *
+ * @example
+ * ```javascript
+ * const validDateInPaste = "2023-05-20";
+ * const invalidDateInFuture = "6078-05-20";
+ *
+ * console.log(isDateInPastOrToday(validDateInPaste)); // true
+ * console.log(isDateInPastOrToday(invalidDateInFuture)); // false
+ * ```
+ */
+export function isDateInPastOrToday(dateString: string): boolean {
+  if (!isValidDateString(dateString)) {
+    throw new Error('Invalid date format. Expected YYYY-MM-DD.');
```

Comment:
Consider throwing an `AppError` here so you can attach an error code.

When we throw AppErrors, the application will automatically catch them and display the error code to the user. The idea is that the user can give this error code to a help desk agent, who will add it to a ticket. 

This will make troubleshooting what went wrong much easier for us.

Reply:
Alternatively, does it make sense to have this function simply return false if the passed-in date is malformed? Maybe not, but it's something worth considering.